.github/workflows/visual-regression-tests.yml

```diff
@@ -0,0 +1,40 @@
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    types:
+      - ready_for_review
+
+jobs:
+  visual-regression-test-components:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "14"
+      - name: Check for cached Ruby gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: ${{ runner.os }}-gems-
+      - name: Get Yarn cache directory path
+        id: yarn-cache-directory-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: Check for cached Yarn dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-directory-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-yarn-
```

Comment:
Ah you've taken to caching the yarn cache over the node_modules directory? Does this offer advantages?  Some googling tells me [it's not recommended](https://stackoverflow.com/a/62244232) to cache node_modules - but I don't understand why? whereas another doc seems to suggest it was better : https://dev.to/mpocock1/how-to-cache-nodemodules-in-github-actions-with-yarn-24eh.

I don't mind which approach we take but I would want want to make sure [our docs](https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-with-github-actions.html#govuk-rails-application-with-postgres-redis-yarn-and-govuk-content-schemas-dependencies) match whichever is preferred.

Again we shouldn't need the os specified here. 

It also reminds me I need to set-up a shared yarn cache for govuk-docker.

Reply:
My understanding is that yarn first tries to install dependencies from the cache folder to the node_modules folder. If the dependency is not cached it's fetched from somewhere on the internet, then cached and installed.

So if the node_modules folder is cached it results in the node_modules folder being overwritten when `yarn install` is run. But if yarn's cache folder is cached, then the that's used to install the dependencies into the node_modules folder - hence this set up.

I can update the documentation with this method if that'd be useful?

Reply:
From a quick play it looks like it checks the node_modules directory first. For example:

```
➜  govuk_publishing_components git:(master) yarn
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 45.96s.
➜  govuk_publishing_components git:(master) yarn cache clean
yarn cache v1.22.10
success Cleared cache.
✨  Done in 2.24s.
➜  govuk_publishing_components git:(master) yarn install
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.30s.
```

I guess using the yarn cache means the cache is a bit broader, ie if PR A removes dependency 1, PR B can still access the cached one. But that seems minor so I'm not really clear on what advantages it gives.

Yup it'd be great to update the documentation to use it, would be great if the commit message justified why it's a better approach so that's recorded.

Reply:
I thought that [`yarn install --frozen-lockfile`](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-pure-lockfile) was the yarn equivalent to [`npm ci`](https://docs.npmjs.com/cli/v6/commands/npm-ci#description). Turns out that `npm ci` removes the node_modules folder before installing dependencies without updating the lockfile, and yarn only installs without updating the lockfile. (Shame that there's no yarn equivalent of `ci` as it's super useful.)

I think it's cleaner to install from yarn's cache as only exact dependencies from the lockfile would be used - caching the node_modules folder could lead to other dependencies being present and other hard to diagnose side effects.

I'll update the examples in the documentation and add the reasoning in the commit message there as well.

Reply:
Ah - I didn't know about `npm ci`. Cool thanks re: updating the docs 👍